webpack.config.js

```diff
@@ -361,4 +304,56 @@ module.exports = (env = process.env.NODE_ENV || 'development') => {
     },
     devtool,
   };
+
+  const mainConfig = {...baseConfig};
+  const previewConfig = {...baseConfig};
+
+  mainConfig.name = 'main';
+  previewConfig.name = 'preview';
+
+  mainConfig.entry = isTest
+    ? undefined
+    : {
+        inline: 'first-input-delay',
+        main: './application.js',
+      };
+  previewConfig.entry = isTest
+    ? undefined
+    : {
+        preview: './preview.js',
+      };
+
+  mainConfig.plugins = [...plugins];
+  previewConfig.plugins = [...plugins];
+
+  if (!isTest) {
+    mainConfig.plugins.push(
+      new OfflinePlugin({
+        caches: {
+          main: [/(?:^|~)(?:main|preview)[-.~]/u, ':externals:'],
+          additional: [':rest:'],
+        },
+        safeToUseOptionalCaches: isProduction,
+        AppCache: {
+          caches: ['main', 'additional', 'optional'],
+        },
+        publicPath: '/',
+        responseStrategy: 'network-first',
+        externals: ['/', 'application.css', 'images/pop/thinking.svg'],
+      }),
+      new HtmlWebpackPlugin({
+        template: './html/index.html',
+        chunksSortMode: 'dependency',
+      }),
+    );
+    previewConfig.plugins.push(
+      new HtmlWebpackPlugin({
+        filename: 'preview.html',
+        template: './html/preview.html',
+        chunksSortMode: 'dependency',
+      }),
+    );
+  }
+
+  return [previewConfig, mainConfig];
```

Comment:
Out of curiosity, can you tell if the main and preview bundles are still sharing certain chunks?

Reply:
I'm not quite sure but It seems that they are not sharing any chunks
preview.html has...
   Entrypoint preview = vendors~preview.js preview.js
index.html has...
    Entrypoint inline = inline.js
    Entrypoint main = vendors~main.js main.js
    a bunch of other chunks

Reply:
Oh well, would be ideal but not a big deal either way. Definitely feels like exporting separate webpack configs is the overall best option considering the limited affordances in `HtmlWebpackPlugin`.